include/node.hpp

```diff
@@ -330,13 +330,20 @@ class node : protected std::tuple<Arguments...> {
         const auto availableForPort            = [&at_least_one_input_has_data]<typename Port>(Port &port) noexcept -> std::pair<std::size_t, std::size_t> {
             std::size_t       availableSamples = port.streamReader().available();
             const std::size_t availableTags    = port.tagReader().available();
+            std::size_t tagsToProcess          = 0;
             if (availableTags > 0) {
                 // at least one tag is present -> if tag is not on the first tag position read up to the tag position
-                auto tagData                  = port.tagReader().get();
+                auto tagData                  = port.tagReader().get(std::min(availableTags, std::size_t{2}));
                 auto tag_stream_head_distance = tagData[0].index - port.streamReader().position();
                 assert(tag_stream_head_distance >= 0 && "negative tag vs. stream distance");
-
-                if (tag_stream_head_distance > 0 && availableSamples > tag_stream_head_distance) {
+                if (tag_stream_head_distance == 0) {
+                    tagsToProcess = 1;
+                    if (availableTags > 1) {
+                        tag_stream_head_distance = tagData[1].index - port.streamReader().position();
+                        assert(tag_stream_head_distance >= 0 && "negative tag vs. stream distance (2nd tag)");
+                    }
+                }
+                if (tag_stream_head_distance > 0 && availableSamples > static_cast<std::size_t>(tag_stream_head_distance)) {
```

Comment:
Note that this code doesn't handle the case yet where there are more than one tag for a sample in the input stream (I think that can happen here, merging seems to happen later?).
Then it would have to get all available streams and iterate/increment tagsToProcess accordingly.